File: backend/app/schemas/candidate.py
"""Candidate schemas - 候选人相关的Pydantic模型"""

from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, EmailStr


class CandidateBase(BaseModel):
    """候选人基础信息"""
    name: str
    phone: Optional[str] = None
    email: Optional[EmailStr] = None
    current_company: Optional[str] = None
    current_title: Optional[str] = None
    years_of_experience: Optional[int] = None
    location: Optional[str] = None
    tags: Optional[List[str]] = None
    source: Optional[str] = None


class CandidateCreate(CandidateBase):
    """创建候选人"""
    pass


class CandidateUpdate(BaseModel):
    """更新候选人"""
    name: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[EmailStr] = None
    current_company: Optional[str] = None
    current_title: Optional[str] = None
    years_of_experience: Optional[int] = None
    location: Optional[str] = None
    tags: Optional[List[str]] = None
    source: Optional[str] = None
    status: Optional[str] = None


class CandidateResponse(CandidateBase):
    """候选人响应"""
    id: str
    status: str
    created_at: datetime
    updated_at: Optional[datetime]
    latest_contacted_at: Optional[datetime]

    class Config:
        from_attributes = True


class CandidateListItem(BaseModel):
    """候选人列表项"""
    id: str
    name: str
    phone: Optional[str]
    email: Optional[str]
    current_company: Optional[str]
    current_title: Optional[str]
    years_of_experience: Optional[int]
    location: Optional[str]
    tags: Optional[List[str]]
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/candidate.ts
export type CandidateStatus = 'new' | 'screening' | 'interviewing' | 'offered' | 'hired' | 'rejected';

export interface Candidate {
  id: string;
  name: string;
  email: string;
  phone: string;
  status: CandidateStatus;
  currentCompany?: string;
  currentTitle?: string;
  yearsOfExperience?: number;
  location?: string;
  avatar?: string;
  tags: string[];
  createdAt: string;
  updatedAt: string;
}

export interface Resume {
  id: string;
  candidateId: string;
  fileName: string;
  fileUrl: string;
  parsedData?: {
    skills: string[];
    education: Array<{
      school: string;
      degree: string;
      major: string;
      startDate: string;
      endDate: string;
    }>;
    experience: Array<{
      company: string;
      title: string;
      startDate: string;
      endDate: string;
      description: string;
    }>;
  };
  uploadedAt: string;
}

export interface Timeline {
  id: string;
  candidateId: string;
  type: 'note' | 'interview' | 'status_change' | 'email' | 'call';
  title: string;
  content: string;
  createdBy: string;
  createdAt: string;
}
